kotlin-mbedtls-netty/src/main/kotlin/org/opencoap/ssl/netty/DtlsChannelHandler.kt

```diff
@@ -114,6 +110,22 @@ class DtlsChannelHandler @JvmOverloads constructor(
         }
     }
 
+    private fun handleDtlsContext(msg: DatagramPacketWithContext, promise: ChannelPromise) {
+        val sessCtx = msg.sessionContext
+        if (sessCtx.sessionSuspensionHint) {
+            promise.toCompletableFuture().thenAccept {
+                dtlsServer.closeSession(msg.recipient())
+            }
+        }
+        if (sessCtx.authenticationContext.isNotEmpty()) {
+            promise.toCompletableFuture().thenAccept {
```

Comment:
I don't think that this needs to run after promise is done. 
Session suspension is a special case.